Ext4Pkg/Ext4Dxe/Superblock.c

```diff
@@ -305,14 +306,16 @@ Ext4OpenSuperblock (
     return EFI_OUT_OF_RESOURCES;
   }
 
-  // Note that the cast below is completely safe, because EXT4_FILE is a specialisation of EFI_FILE_PROTOCOL
-  Status = Ext4OpenVolume (&Partition->Interface, (EFI_FILE_PROTOCOL **)&Partition->Root);
+  Status = Ext4OpenVolume (&Partition->Interface, &PartitionRootProtocol);

```

Comment:
Why?

Reply:
Hrm, I may get why. If CR() messes with the structure offsets, it's more insane than I thought

Reply:
You should not use direct casts anyway because updates to the structure may break the code with no warning at all. Using `BASE_CR` is free computation-wise and safe against bugs introduced by structure rearrangement.